test/e2e/suites/unmanaged/unmanaged_CAPI_test.go

```diff
@@ -73,9 +73,9 @@ var _ = ginkgo.Context("[unmanaged] [Cluster API Framework]", func() {
 			Expect(shared.AcquireResources(requiredResources, config.GinkgoConfig.ParallelNode, flock.New(shared.ResourceQuotaFilePath))).To(Succeed())
 		})
 
-		capi_e2e.MachinePoolSpec(ctx, func() capi_e2e.MachinePoolInput {
-			return capi_e2e.MachinePoolInput{
-				E2EConfig:             e2eCtx.E2EConfig,
+		MachinePoolSpec(ctx, func() MachinePoolInput {
```

Comment:
I understand why you decided to create a separate MachinePoolSpec instead of using CAPI framework, because [workerMachineCount](https://github.com/kubernetes-sigs/cluster-api/blob/6b042456383aa18ba61fc5de5c41901dd7267c60/test/e2e/machine_pool.go#L78) is hardcoded and not configurable.

It's ok for now but I think we should raise a PR to CAPI to make workerMachineCount configurable similar to [ClusterUpgradeConformanceSpec](https://github.com/kubernetes-sigs/cluster-api/blob/main/test/e2e/cluster_upgrade.go#L52) so that we don't need to keep a separate copy of MachinePool spec.

Reply:
I agree and making it configurable makes a lot of sense. 
Due to the issue with the controller not being able to reconcile a MachinePool with zero replicas, I opened kubernetes-sigs/cluster-api#6210 and started working on a [PR (diff)](https://github.com/kubernetes-sigs/cluster-api/compare/main...helio:machinepool-empty-providerIDList?expand=1). I took a bit of a different route with adjusting the MachinePoolSpec, by explicitly adding a scale to zero step. 

Not sure what's the best approach there, I guess it depends on what cases the e2e test should cover. I'd be glad for a review anyway. I wasn't able to run this change together with the changes in CAPI yet, so that's still an open task for me. 

I've also seen that the tests ran into some linting issue - apologies, I didn't check those before submitting.

Reply:
The approach of adding another step to MachinePool Spec to scale down replicas to 0 sounds good to me. Let's see what CAPI folks say when the PR is submitted. Thanks for working on this.

Reply:
FYI kubernetes-sigs/cluster-api#6233 is submitted which is required as a base for this. Will update this PR once merged :crossed_fingers:

Reply:
As https://github.com/kubernetes-sigs/cluster-api/pull/6233 was merged in CAPI 🎉, we can remove this file. 
Once [CAPA updates to use CAPI v1.1.3](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3292), e2e test will test minSize case of 0.